lms/djangoapps/courseware/user_state_client.py

```diff
@@ -193,6 +193,11 @@ def set_many(self, username, block_keys_to_state, scope=Scope.user_state):
         else:
             user = User.objects.get(username=username)
 
+        if user.is_anonymous():
```

Comment:
I think in this case, checking for `AnonymousUser` is the right thing. The `known` distinction is really for cases where the XBlock wants to lazy-load the user, so we need to bypass an auth check in order to give it a chance to do that. The `AnonymouseUser` check here is just because we know, if we have an `AnonymousUser` here, that we can't possibly save their state to the database (because we have no identifier to save it against).